memory/fork_mem.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Harish <harish@linux.vnet.ibm.com>
+#
+
+
+import os
+import shutil
+from avocado import Test
+from avocado import main
+from avocado.utils import process, build, memory
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Forkoff(Test):
+    """
+    Perform mmap and process forking to maxiumum extent
+
+    :avocado: tags=memory
+    """
+
+    def setUp(self):
+        '''
+        Use 90% of memory with/without many process forked
+        '''
+
+        smm = SoftwareManager()
+        self.iter = self.params.get('iterations', default='10')
+        self.procs = self.params.get('procs', default='1')
+        self.minmem = self.params.get('minmem', default='10')
+
+        self.freemem = int((0.9 * memory.freememtotal()) / 1024)
+        # Check for basic utilities
+        for packages in ['gcc', 'make']:
+            if not smm.check_installed(packages) and not smm.install(packages):
+                self.cancel('%s is needed for the test to be run' % packages)
+
+        shutil.copyfile(os.path.join(self.datadir, 'forkoff.c'),
+                        os.path.join(self.teststmpdir, 'forkoff.c'))
+
+        shutil.copyfile(os.path.join(self.datadir, 'Makefile'),
+                        os.path.join(self.teststmpdir, 'Makefile'))
+
+        build.make(self.teststmpdir)
+
+    def test(self):
+        '''
+        Execute memory fork off tests
+        1. Total memory with one process
+        2. Split memeory between given processes
+        3. Maximum process with minimum memory (10 MB) per process
+        '''
+        os.chdir(self.teststmpdir)
+
+        if process.system('./forkoff %s %s %s'
+                          % (self.freemem, 1, self.iter),
+                          shell=True, ignore_status=True) != 0:
+            self.fail("Test failed")
```

Comment:
Would you want to fail fast or try all combinations and fail in the last ?

Reply:
All combinations would help, will change accordingly.